File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/DialogLevel.java
package com.nopalsoft.sokoban.scene2d;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.Stage;
import com.badlogic.gdx.scenes.scene2d.ui.Button;
import com.badlogic.gdx.scenes.scene2d.ui.Image;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.Label.LabelStyle;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.nopalsoft.sokoban.Assets;
import com.nopalsoft.sokoban.game.GameScreen;
import com.nopalsoft.sokoban.screens.Screens;

public class DialogLevel extends Dialog {

    Button btPlay;
    Label labelBestMoves, labelBestTime;

    public DialogLevel(Screens currentScreen) {
        super(currentScreen, 350, 300, 100);

        setCloseButton();
        setTitle("Puntuaciones", .75f);

        Table tbMenu = new Table();
        tbMenu.setFillParent(true);

        btPlay = new Button(Assets.btPlay, Assets.btPlayPress);

        Image imgClock = new Image(Assets.clock);
        Image imgMoves = new Image(Assets.playerStand);

        labelBestMoves = new Label("0", new LabelStyle(Assets.fontRed, Color.WHITE));
        labelBestTime = new Label("0", new LabelStyle(Assets.fontRed, Color.WHITE));

        tbMenu.defaults().expandX();

        tbMenu.padLeft(30).padRight(30).padBottom(20).padTop(50);
        tbMenu.add(imgMoves).size(45);
        tbMenu.add(labelBestMoves);

        tbMenu.row().padTop(10);
        tbMenu.add(imgClock).size(45);
        tbMenu.add(labelBestTime);

        tbMenu.row().padTop(10);
        tbMenu.add(btPlay).colspan(2).size(60);

        addActor(tbMenu);

    }

    public void show(Stage stage, final int level, int bestMoves, int bestTime) {
        labelBestMoves.setText(bestMoves + "");
        labelBestTime.setText(bestTime + "");

        btPlay.clear();
        btPlay.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                screen.changeScreenWithFadeOut(GameScreen.class, level, screen.game);
            }
        });

        super.show(stage);
    }

}

File: libgdx_sokoban/core/src/com/nopalsoft/sokoban/scene2d/DialogLevel.kt
package com.nopalsoft.sokoban.scene2d

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.scenes.scene2d.InputEvent
import com.badlogic.gdx.scenes.scene2d.Stage
import com.badlogic.gdx.scenes.scene2d.ui.Button
import com.badlogic.gdx.scenes.scene2d.ui.Image
import com.badlogic.gdx.scenes.scene2d.ui.Label
import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener
import com.nopalsoft.sokoban.Assets
import com.nopalsoft.sokoban.game.GameScreen
import com.nopalsoft.sokoban.screens.Screens

/**
 * The dialog that will be shown whenever you choose a game from the main screen.
 * This dialog shows the best number of  moves and the best time that the player
 * has had in that specific level.
 */
class DialogLevel(currentScreen: Screens) : Dialog(currentScreen, 350F, 300F, 100F) {

    private val buttonPlay = Button(Assets.btPlay, Assets.btPlayPress);
    private val labelBestMoves = Label("0", Label.LabelStyle(Assets.fontRed, Color.WHITE))
    private val labelBestTime = Label("0", Label.LabelStyle(Assets.fontRed, Color.WHITE))

    init {
        setCloseButton()
        setTitle("Puntuaciones", .75f)

        val tableMenu = Table()
        tableMenu.setFillParent(true)

        val imageClock = Image(Assets.clock)
        val imageMoves = Image(Assets.playerStand)

        tableMenu.defaults().expandX()

        tableMenu.padLeft(30f).padRight(30f).padBottom(20f).padTop(50f)
        tableMenu.add(imageMoves).size(45f)
        tableMenu.add(labelBestMoves)

        tableMenu.row().padTop(10f)
        tableMenu.add(imageClock).size(45f)
        tableMenu.add(labelBestTime)

        tableMenu.row().padTop(10f)
        tableMenu.add(buttonPlay).colspan(2).size(60f)

        addActor(tableMenu)
    }

    fun show(stage: Stage?, level: Int, bestMoves: Int, bestTime: Int) {
        labelBestMoves.setText(bestMoves.toString())
        labelBestTime.setText(bestTime.toString())

        buttonPlay.clear()
        buttonPlay.addListener(object : ClickListener() {
            override fun clicked(event: InputEvent, x: Float, y: Float) {
                screen.changeScreenWithFadeOut(GameScreen::class.java, level, screen.game)
            }
        })

        super.show(stage)
    }
}